sopel/config/core_section.py

```diff
@@ -111,120 +139,271 @@ class CoreSection(StaticSection):
+    """
 
     channels = ListAttribute('channels')
     """List of channels for the bot to join when it connects."""
 
-    db_type = ChoiceAttribute('db_type', choices=[
-        'sqlite', 'mysql', 'postgres', 'mssql', 'oracle', 'firebird', 'sybase'], default='sqlite')
-    """The type of database to use for Sopel's database.
+    db_driver = ValidatedAttribute('db_driver')
+    """The driver to use for connecting to the database.
 
-    mysql - pip install mysql-python (Python 2) or pip install mysqlclient (Python 3)
-    postgres - pip install psycopg2
-    mssql - pip install pymssql
+    This is optional, but can be specified if user wants to use a different
+    driver than the default for the chosen :attr:`db_type`.
+
+    .. seealso::
+
+        Refer to :ref:`SQLAlchemy's documentation <engines_toplevel>` for more
+        information.
 
-    See https://docs.sqlalchemy.org/en/latest/dialects/ for a full list of
-    dialects.
     """
 
     db_filename = ValidatedAttribute('db_filename')
-    """The filename for Sopel's database. (SQLite only)"""
-
-    db_driver = ValidatedAttribute('db_driver')
-    """The driver for Sopel's database.
+    """The filename for Sopel's database.
 
-    This is optional, but can be specified if user wants to use a different
-    driver.
+    Used only for SQLite. Ignored for all other :attr:`db_type` values.
+    """
 
-    .. seealso::
+    db_host = ValidatedAttribute('db_host')
+    """The host for Sopel's database.
 
-        https://docs.sqlalchemy.org/en/latest/core/engines.html
+    Ignored when using SQLite.
     """
 
-    db_user = ValidatedAttribute('db_user')
-    """The user for Sopel's database."""
+    db_name = ValidatedAttribute('db_name')
+    """The name of Sopel's database.
+
+    Ignored when using SQLite.
+    """
 
     db_pass = ValidatedAttribute('db_pass')
-    """The password for Sopel's database."""
+    """The password for Sopel's database.
 
-    db_host = ValidatedAttribute('db_host')
-    """The host for Sopel's database."""
+    Ignored when using SQLite.
+    """
 
     db_port = ValidatedAttribute('db_port')
-    """The port for Sopel's database."""
+    """The port for Sopel's database.
 
-    db_name = ValidatedAttribute('db_name')
-    """The name of Sopel's database."""
+    Ignored when using SQLite.
+    """
+
+    db_type = ChoiceAttribute('db_type', choices=[
+        'sqlite', 'mysql', 'postgres', 'mssql', 'oracle', 'firebird', 'sybase'], default='sqlite')
+    """The type of database Sopel should connect to.
+
+    :default: ``sqlite`` (part of Python's standard library)
+
+    The full list of values Sopel recognizes is:
+
+    * ``firebird``
+    * ``mssql``
+    * ``mysql``
+    * ``oracle``
+    * ``postgres``
+    * ``sqlite``
+    * ``sybase``
+
+    Here are the additional PyPI packages you may need to install to use one of
+    the most commonly requested alternatives:
+
+    mysql
+      ``pip install mysql-python`` (Python 2)
+
```

Comment:
Does this empty line is required?

Reply:
Unfortunately, yes. Without it, Sphinx considers this as a single wrapped paragraph:

![image](https://user-images.githubusercontent.com/164140/70412526-99cc5b00-1a1a-11ea-9e2c-f97192c51416.png)

I wasn't able to find a way of forcing a line-break in rST. :/

Would you rather see each term (`mysql`, `postgres`, `mssql`) preceded by a blank line? That still generates a clean `<dl>` but looks slightly better in source-code form.

Reply:
The more you know. Thanks for your answer! :+1: